packages/kit/src/runtime/client/start.js

```diff
@@ -3,17 +3,18 @@ import { create_client } from './client.js';
 import { init } from './singletons.js';
 
 /**
- * @param {import('./types').SvelteKitApp} app
  * @param {HTMLElement} target
  * @param {Parameters<import('./types').Client['_hydrate']>[0]} [hydrate]
  */
-export async function start(app, target, hydrate) {
+export async function start(target, hydrate) {
 	if (DEV && target === document.body) {
 		console.warn(
 			`Placing %sveltekit.body% directly inside <body> is not recommended, as your app may break for users who have certain browser extensions installed.\n\nConsider wrapping it in an element:\n\n<div style="display: contents">\n  %sveltekit.body%\n</div>`
 		);
 	}
 
+	// @ts-expect-error
+	const app = await import('__sveltekit/APP');
```

Comment:
Can we achieve the desired outcome differently? I believe this makes the client.js bundle less cacheable between builds

Reply:
In theory, `start` shouldn't change between builds, so it should always be cached after the initial visit, even after a redeploy. `app` on the other hand will change frequently (though less frequently with #4482). So if we combine them into a single chunk, the app as a whole gets less cacheable. The current split is very deliberate, for that reason, so we should probably close this PR.

Having said that, I just tried building an app twice, and if `kit.version.name` isn't fixed it _will_ result in different `start` chunks. I thought we'd fixed that a while back but apparently not. We should investigate.

Reply:
You don't need to use a dynamic import for that IIUC, you just need to configure Rollup appropriately — if module A statically imports module B, and module B is an entry point, it will work as expected.

Bear in mind that this would create a waterfall for browsers that don't implement `modulepreload`. I'm not sure that's a change we want to make. It strikes me that if we want to have a non-code-split mode we'll want a separate bundling strategy specifically for that case